pkg/secrethub/repo.go

```diff
@@ -309,3 +316,113 @@ func (c *Client) getRepoIndexKey(repoPath api.RepoPath) (*crypto.SymmetricKey, e
 
 	return repoIndexKey, nil
 }
+
+// Iterator returns a new iterator that retrieves all repos in the given namespace.
+func (s repoService) Iterator(namespace string, params *RepoIteratorParams) RepoIterator {
```

Comment:
We also discussed having iteration of repos in a given namespace and iteration of all repos you have access to done in one iterator. Why did you choose for separating them?

Reply:
Because if namespace would be an optional param in `RepoIteratorParams` then this set of parameters would not make sense in the context of the `MeService`'s `RepoIterator` method. I think it is better if the `RepoIteratorParams` are the params used whenever iterating over repos regardless of service.

Reply:
> Because if namespace would be an optional param in RepoIteratorParams then this set of parameters would not make sense in the context of the MeService's RepoIterator method.

Why not? The iterator in the me service is returning all repositories the authenticated user has access to. The namespace parameter could limit the response to those repositories that are in the given namespace.
You can see it as a filter.

Reply:
Okay that makes sense, apparently the functionality of iterator in the me service wasn't clear to me.